src/sms/config.py

```diff
@@ -0,0 +1,108 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Config Parameter Modeling and Parsing."""
+
+from re import split
+
+from ghga_service_commons.api import ApiConfigBase
+from hexkit.config import config_from_yaml
+from hexkit.log import LoggingConfig
+from pydantic import Field, SecretStr, field_validator
+from pydantic_settings import BaseSettings
+
+SERVICE_NAME: str = "sms"
+
+
+class SmsConfig(BaseSettings):
+    """Configuration specific to the SMS."""
+
+    token_hashes: list[str] = Field(
+        default=...,
+        description="List of token hashes corresponding to the tokens that can be used "
+        + "to authenticate calls to this service. Hashes are made with SHA-256.",
+        examples=["7ad83b6b9183c91674eec897935bc154ba9ff9704f8be0840e77f476b5062b6e"],
+    )
+    db_prefix: str = Field(
+        default=...,
+        description="Prefix to add to all database names used in the SMS.",
+        examples=["testing-branch-name-"],
+    )
+    db_permissions: list[str] = Field(
+        default=[],
+        description=(
+            "List of permissions that can be granted on a collection. Use * to signify"
+            + " 'all'. The format is '<db_name>.<collection_name>:<permissions>', e.g."
+            + " 'db1.collection1.crud'. The permissions are 'r' for read and 'w' for"
+            + " write. '*' can be used to mean both read and write (or 'rw')."
+            + " Deletion is a write operation. If db_permissions are not set, no"
+            + " operations are allowed on any database or collection."
+        ),
+        examples=[
+            "db1.coll1:r",
+            "db1.coll1:w",
+            "db1.coll1:rw",
+            "db1.coll1:*",
+            "db2.*:r",
+            "db3.*:*",
+            "*.*:r",
+            "*.*:*",
+        ],
+    )
+    db_connection_str: SecretStr = Field(
+        default=...,
+        examples=["mongodb://localhost:27017"],
+        description=(
+            "MongoDB connection string. Might include credentials."
+            + " For more information see:"
+            + " https://naiveskill.com/mongodb-connection-string/"
+        ),
+    )
+
+    @field_validator("db_permissions")
+    @classmethod
+    def validate_permissions(cls, value):
+        """Validate the permissions to make sure only 'r', 'w', 'rw', and '*' are used.
+
+        Rules are normalized by unpacking '*' into 'rw'.
+        """
+        if value is None:
+            return value
+
+        normalized_rules: list[str] = []
+        for permission in value:
+            db_name, collection, ops = split(r"\.(.*?):", permission)
```

Comment:
I don't think re.split is meant to be used with such a combined separator pattern, and it probably will break if the collection name contains a ":". I would use a normal split here which is much easier to understand:
```
try:
    namespace, ops = permission.rsplit(':' , 1)
    db_name, collection = permission.split('.' , 1)
    if not (db_name and collection and ops):
        raise ValueError
except ValueError:
    raise ValueError("... must have the format ...")
```

Reply:
You're right, I didn't consider that. See [f71217c](https://github.com/ghga-de/state-management-service/pull/1/commits/f71217c95008826977207d0f7c25f20665896c95)